src/shared-render.test.js

```diff
@@ -0,0 +1,93 @@
+const {h, Component} = require('preact');
+
+const {render, shallow} = require('./preact-render-spy');
+
+const shape = (func, funcMember) => {
+  it(`${funcMember}: renders into fragment`, async () => {
+    class Node extends Component {
+      render() {
+        return <div />;
+      }
+    }
+    const context = func(<Node />);
+    expect(context.fragment.children.length).toBe(1);
+    expect(context.fragment.children[0].tagName).toBe('DIV');
+    expect(context.find('div').length).toBe(1);
+    expect(context.find('div').contains(<div />)).toBeTruthy();
+  });
+
+  it(`${funcMember}: renders props`, async () => {
+    class Node extends Component {
+      render({children}) {
+        return <div>{children}</div>;
+      }
+    }
+    const context = func(<Node>node</Node>);
+    expect(context.find('div').length).toBe(1);
+    expect(context.find('div').text('node')).toBeTruthy();
+  });
+
+  it(`${funcMember}: renders changes`, async () => {
+    class Node extends Component {
+      render({children}) {
+        return <div>{children}</div>;
+      }
+    }
+    const context = func(<Node>node</Node>);
+    context[funcMember](<Node>node2</Node>);;
+    expect(context.find('div').contains('node2')).toBeTruthy();
+  });
+
+  it(`${funcMember}: renders change on click`, async () => {
+    class Node extends Component {
+      constructor(...args) {
+        super(...args);
+
+        this.state = {count: 0};
+        this.onClick = this.onClick.bind(this);
+      }
+
+      onClick() {
+        this.setState({count: this.state.count + 1});
+      }
+
+      render({}, {count}) {
+        return <div onClick={this.onClick}>{count}</div>;
+      }
+    }
+    const context = func(<Node/>);
+    expect(context.find('div').contains('0')).toBeTruthy();
+    context.find('div').simulate('click');
+    expect(context.find('div').contains('1')).toBeTruthy();
+  });
+
+  it(`${funcMember}: renders multiple components`, () => {
+    class Node extends Component {
+      render({count}) {
+        return <div>{count}</div>;
+      }
+    }
+    const context = func(<div><Node count="1" /><Node count="2" /></div>);
+    expect(context.find('div').contains('1')).toBeTruthy();
+    expect(context.find('div').contains(<div>1</div>)).toBeTruthy();
+    expect(context.find('div').contains('2')).toBeTruthy();
+    expect(context.find('div').contains(<div>2</div>)).toBeTruthy();
+  });
+
+  it(`${funcMember}: renders stateless components`, () => {
+    const Node = ({count}) => <div>{count}</div>;
+    const context = func(<Node count="1" />);
+    expect(context.find('div').length).toBe(1);
+    expect(context.find('div').contains('1')).toBeTruthy();
+  });
+
+  it(`${funcMember}: renders components with null children`, () => {
```

Comment:
not that it's a big deal, but since `shallow` wouldn't ever even render `Empty` I think we should potentially add a test where just `Node = () => null;` and `context = func(<div><Node />text</div>)`